rust/feature-flags/src/handler/flags.rs

```diff
@@ -10,29 +10,181 @@ use crate::{
     },
 };
 use axum::extract::State;
+use serde::{Deserialize, Serialize};
 use serde_json::Value;
 use std::collections::HashMap;
 use uuid::Uuid;
 
 use super::{evaluation, types::FeatureFlagEvaluationContext};
 use crate::router;
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum EvaluationRuntime {
+    All,
+    Client,
+    Server,
+}
+
+impl From<String> for EvaluationRuntime {
+    fn from(s: String) -> Self {
+        EvaluationRuntime::from(s.as_str())
+    }
+}
+
+impl From<&str> for EvaluationRuntime {
+    fn from(s: &str) -> Self {
+        if s.eq_ignore_ascii_case("client") {
+            EvaluationRuntime::Client
+        } else if s.eq_ignore_ascii_case("server") {
+            EvaluationRuntime::Server
+        } else {
+            EvaluationRuntime::All
+        }
+    }
+}
+
+/// Determines the evaluation runtime based on request characteristics.
+/// Uses explicit runtime if provided, otherwise analyzes user-agent, headers, and lib_version
+/// to detect if the request is from a client-side (browser/mobile) or server-side SDK.
+fn detect_evaluation_runtime_from_request(
+    headers: &axum::http::HeaderMap,
+    lib_version: Option<&str>,
+    explicit_runtime: Option<EvaluationRuntime>,
+) -> Option<EvaluationRuntime> {
+    // Use explicitly passed runtime if available
+    if let Some(runtime) = explicit_runtime {
+        return Some(runtime);
+    }
+
+    // Analyze User-Agent header
+    if let Some(user_agent) = headers.get("user-agent").and_then(|v| v.to_str().ok()) {
+        // Browser patterns - these typically indicate client-side execution
+        if user_agent.contains("Mozilla/")
+            || user_agent.contains("Chrome/")
+            || user_agent.contains("Safari/")
+            || user_agent.contains("Firefox/")
+            || user_agent.contains("Edge/")
+        {
+            return Some(EvaluationRuntime::Client);
+        }
+
+        // Server SDK patterns - these indicate server-side execution
+        if user_agent.starts_with("posthog-python/")
+            || user_agent.starts_with("posthog-ruby/")
+            || user_agent.starts_with("posthog-php/")
+            || user_agent.starts_with("posthog-java/")
```

Comment:
Unfortunately, `posthog-android` sends its user agent as `posthog-java/` which means this would send Android server flags instead of client.

1. [This is where the user-agent is set to `config.userAgent`](https://github.com/PostHog/posthog-android/blob/06a5ba0ccb90bdec5f1697665498bee65fb72ac7/posthog/src/main/java/com/posthog/internal/PostHogApi.kt#L110)
2. [This is where `config.userAgent` is set to `"$sdkName/$sdkVersion"`](https://github.com/PostHog/posthog-android/blob/06a5ba0ccb90bdec5f1697665498bee65fb72ac7/posthog/src/main/java/com/posthog/PostHogConfig.kt#L177-L180)
3. [And this is where `sdkName` is set to `posthog-java`](https://github.com/PostHog/posthog-android/blob/06a5ba0ccb90bdec5f1697665498bee65fb72ac7/posthog/src/main/java/com/posthog/PostHogConfig.kt#L172)

Reply:
nope, posthog-android sets posthog-android/version as user agent
https://github.com/PostHog/posthog-android/blob/06a5ba0ccb90bdec5f1697665498bee65fb72ac7/posthog-android/src/main/java/com/posthog/android/PostHogAndroid.kt#L96-L97

Reply:
Oh good! Thanks for chiming in.